monitor.py

```diff
@@ -866,39 +866,45 @@ def parse_entry(self, line: bytes) -> MutableMapping[str, Any]:  # noqa: C901, C
 
             elif event_type == 'shiplocker':
                 # As of 4.0.0.400 (2021-06-10)
-                # "ShipLocker" will be a full list written to the journal at startup/boarding/disembarking, and also
+                # "ShipLocker" will be a full list written to the journal at startup/boarding, and also
                 # written to a separate shiplocker.json file - other updates will just update that file and mention it
                 # has changed with an empty shiplocker event in the main journal.
 
-                # Always attempt loading of this.
-                # Confirmed filename for 4.0.0.400
-                try:
-                    currentdir_path = pathlib.Path(str(self.currentdir))
-                    with open(currentdir_path / 'ShipLocker.json', 'rb') as h:  # type: ignore
-                        entry = json.load(h, object_pairs_hook=OrderedDict)
-                        self.state['ShipLockerJSON'] = entry
+                # Always attempt loading of this, but if it fails we'll hope this was
+                # a startup/boarding version and thus `entry` contains
+                # the data anyway.
+                currentdir_path = pathlib.Path(str(self.currentdir))
+                shiplocker_filename = currentdir_path / 'ShipLocker.json'
+                shiplocker_max_attempts = 5
+                shiplocker_fail_sleep = 0.01
+                attempts = 0
+                while attempts < shiplocker_max_attempts:
+                    attempts += 1
+                    try:
+                        with open(shiplocker_filename, 'rb') as h:  # type: ignore
+                            entry = json.load(h, object_pairs_hook=OrderedDict)
+                            self.state['ShipLockerJSON'] = entry
+                            break
 
-                except FileNotFoundError:
-                    logger.warning('ShipLocker event but no ShipLocker.json file')
-                    pass
+                    except FileNotFoundError:
+                        logger.warning('ShipLocker event but no ShipLocker.json file')
+                        sleep(shiplocker_fail_sleep)
+                        pass
 
-                except json.JSONDecodeError as e:
-                    logger.warning(f'ShipLocker.json failed to decode:\n{e!r}\n')
-                    pass
+                    except json.JSONDecodeError as e:
+                        logger.warning(f'ShipLocker.json failed to decode:\n{e!r}\n')
+                        sleep(shiplocker_fail_sleep)
+                        pass
 
```

Comment:
```suggestion
                else:
                    logger.warning(f'Failed to decode shiplocker after {shiplocker_max_attempts} tries. Giving up.')

```

Reply:
thats awful. thanks github. The suggestion is to add an else to the while, to log when its not broken out of

Reply:
It took me a moment looking at it in email, but I fathomed what you meant ;).